### ISSUE TITLE
Maintain and document a list of equivalent messages in other tools

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [pylint-dev/pylint#8579]
### Current problem

There's overlap with other tools message, and pylint is slower. (If ruff does it and fix it then you probably don't want to check a message in pylint). There's also the possibiity to use existing noqa so using multiple linter is easier.

### Desired solution

A wish list of thing I thought about but that are open for discussion of course:

- [ ] A data structure linking equivalent messages from pylint to message from other tools (wrong import position -> message in isort / ruff)
- [ ] Inclusion of this information in our documentation (beside the manually generated https://pylint.readthedocs.io/en/latest/faq.html#which-messages-should-i-disable-to-avoid-duplicates-if-i-use-other-popular-linters)
- [ ] Auto-populated from what is implemented in ruff, in ruff case (See script in https://github.com/charliermarsh/ruff/pull/2546)
- [ ] An indication of the possibility to autofix with other tools (isort and ruff autofix wrong-import-order for example)
- [ ] A check that the result is equivalent in our functional tests
- [ ] Adding this information in our doc if the check is not equivalent
- [ ] Check that the autofix from other tools if they exists actually fix the problem in our functional test file
- [ ] Adding this information in our doc if the autofix is incomplete
- [ ] disable/noqa from other tools, automatically disabling pylint's own messages

### Additional context

https://github.com/pylint-dev/pylint/issues/2493
https://github.com/charliermarsh/ruff/pull/2546
https://pylint.readthedocs.io/en/latest/faq.html#which-messages-should-i-disable-to-avoid-duplicates-if-i-use-other-popular-linters

**Comment by Pierre-Sassoulas** [comment]
Here's some manual work done by @antonagestam in https://github.com/charliermarsh/ruff/issues/970#issuecomment-1565594417: https://github.com/antonagestam/pylint-mypy-overlap

**Comment by antonagestam** [comment]
@Pierre-Sassoulas Perhaps it'd even be possible to introduce profiles, so that users can define what other tools they're using, and that would disable a set of checks?

**Comment by Pierre-Sassoulas** [comment]
Yeah, that would need https://github.com/pylint-dev/pylint/issues/7120 though.
